apps/service_providers/migrations/0020_auto_20241111_1113.py

```diff
@@ -0,0 +1,264 @@
+        )
+        analysis.llm_provider_model = global_provider_model
+        analysis.save()
+        return
+    except LlmProviderModel.DoesNotExist:
+        pass
+
+    try:
+        custom_provider_model = LlmProviderModel.objects.get(
+            team=analysis.team,
+            type=analysis.llm_provider.type,
+            name=analysis.llm_model,
+        )
+        analysis.llm_provider_model = custom_provider_model
+        analysis.save()
+        return
+    except LlmProviderModel.DoesNotExist:
+        pass
+
+    new_custom_provider_model = LlmProviderModel.objects.create(
+        team=analysis.team,
+        type=analysis.llm_provider.type,
+        name=analysis.llm_model,
+    )
+    analysis.llm_provider_model = new_custom_provider_model
+    analysis.save()
+
+def _handle_assistant(LlmProviderModel, assistant):
+    try:
+        global_provider_model = LlmProviderModel.objects.get(
+            team__isnull=True,
+            type=assistant.llm_provider.type,
+            name=assistant.llm_model,
+        )
+        assistant.llm_provider_model = global_provider_model
+        assistant.save()
+        return
+    except LlmProviderModel.DoesNotExist:
+        pass
+
+    try:
+        custom_provider_model = LlmProviderModel.objects.get(
+            team=assistant.team,
+            type=assistant.llm_provider.type,
+            name=assistant.llm_model,
+        )
+        assistant.llm_provider_model = custom_provider_model
+        assistant.save()
+        return
+    except LlmProviderModel.DoesNotExist:
+        pass
+
+    new_custom_provider_model = LlmProviderModel.objects.create(
+        team=assistant.team,
+        type=assistant.llm_provider.type,
+        name=assistant.llm_model,
+    )
+    assistant.llm_provider_model = new_custom_provider_model
+    assistant.save()
+
+def _handle_llm_experiment(LlmProviderModel, experiment):
+    try:
+        global_provider_model = LlmProviderModel.objects.get(
+            team__isnull=True,
+            type=experiment.llm_provider.type,
+            name=experiment.llm,
+            max_token_limit=experiment.max_token_limit_old
+        )
+        experiment.llm_provider_model = global_provider_model
+        experiment.save()
+        return
+    except LlmProviderModel.DoesNotExist:
+        pass
+
+    try:
+        custom_provider_model = LlmProviderModel.objects.get(
+            team=experiment.team,
+            type=experiment.llm_provider.type,
+            name=experiment.llm,
+            max_token_limit=experiment.max_token_limit_old
+        )
+        experiment.llm_provider_model = custom_provider_model
+        experiment.save()
+        return
+    except LlmProviderModel.DoesNotExist:
+        pass
+
+    new_custom_provider_model = LlmProviderModel.objects.create(
+        team=experiment.team,
+        type=experiment.llm_provider.type,
+        name=experiment.llm,
+        max_token_limit=experiment.max_token_limit_old,
+    )
```

Comment:
Yes, I was going to refactor these, but they are different enough (e.g. `experiment` has the `max_token_limit` check) that I thought it was OK to leave as-is, especially since this code will be run only once. 
What do you think?